specs/merge/validator.md

```diff
@@ -37,18 +38,38 @@ Please see related Beacon Chain doc before continuing and use them as a referenc
 
 ### `ExecutionEngine`
 
-The following methods are added to the `ExecutionEngine` protocol for use as a validator:
+*Note*: `prepare_payload` and `get_payload` functions are added to the `ExecutionEngine` protocol for use as a validator.
 
-#### `assemble_block`
+The body of each of these functions is implementation dependent.
+The Engine API may be used to implement them with an external execution engine.
 
-Produces a new instance of an execution payload, with the specified `timestamp`,
-on top of the execution payload chain tip identified by `block_hash`.
+#### `prepare_payload`
 
-The body of this function is implementation dependent.
-The Consensus API may be used to implement this with an external execution engine.
+Given the set of execution payload attributes, `prepare_payload` initiates a process of building an execution payload
+on top of the execution chain tip identified by `parent_hash`.
 
 ```python
-def assemble_block(self: ExecutionEngine, block_hash: Hash32, timestamp: uint64, random: Bytes32) -> ExecutionPayload:
+def prepare_payload(self: ExecutionEngine,
+                    parent_hash: Hash32,
+                    timestamp: uint64,
+                    random: Bytes32,
+                    fee_recipient: Bytes20) -> uint64:
+    """
+    Returns ``payload_id`` that is used to obtain the execution payload in a subsequent ``get_payload`` call.
+    """
+    ...
+```
+
+#### `get_payload`
+
+Given the `payload_id`, `get_payload` returns the most recent version of the execution payload that
```

Comment:
I don't think it should be in the spec as it's implementation dependent. In the Engine API specification such call to this method is responded with error. But in some other implementation e.g. a client having both layers in one unit of software it could do anything else, like responding with empty payload or so.